File: programs/extending-classes_player-class-4.cpp
Card* drawForMatch(Eights* eights, const Card& prev) {
  while (true) {
    Card* card = new Card(eights->drawCard());
    cout << name << " draws " << *card << endl;
    if (cardMatches(*card, prev)) {
      return card;
    }
    hand.addCard(*card);
    delete card; // Avoid memory leak
  }
}

File: programs/extending-classes_player-class-4.java
public Card drawForMatch(Eights eights, Card prev) {
    while (true) {
        Card card = eights.drawCard();
        System.out.println(name + " draws " + card);
        if (cardMatches(card, prev)) {
            return card;
        }
        hand.addCard(card);
    }
}